test/components/form/form_test.exs

```diff
@@ -0,0 +1,71 @@
+defmodule Surface.Components.Form.FormTest do
+  use ExUnit.Case
+
+  alias Surface.Components.Form.{
+          Form,
+          TextInput
+        },
+        warn: false
+
+  import ComponentTestHelper
+
+  defmodule ViewWithForm do
+    use Surface.LiveView
+
+    def render(assigns) do
+      ~H"""
+      <Form for={{ @changeset }} action="#" opts={{ csrf_token: "test", as: :user }}>
+        <TextInput field="name" />
+      </Form>
+      """
+    end
+
+    def mount(_params, session, socket) do
+      {:ok, assign(socket, changeset: session["changeset"])}
+    end
+  end
+
+  test "form as an atom" do
```

Comment:
Sweet! I didn't know we could pass an atom. Since we can, how about testing only this case and get rid of the dependencies related to ecto? I mean, the changeset stuff is already tested by phoenix itself. WDYT?

Reply:
That's more or less true, but at the same time the way a changeset interacts with the other inputs is different than the atom. For example, I only realized that we can't see the value to `nil` in the text input when testing with the changeset, because the default prop was overriding the changeset changes.

Also, isn't it only a test dependency? Does it get installed when people include the library?

Reply:
It's a test only dependency indeed. I just want to make sure we're not testing things that are not directly related to our implementation. The question is: Do we have any logic implemented that is specific to changesets? If so, then we should test it for sure, otherwise, we just have to make sure we call the wrapped function correctly and trust that it works. Feel free to keep the dependency if you think that's the scenario and also in case you foresee other components falling into this category.

Reply:
I’m definitely going to test other components with changesets to ensure all is working properly. However the idea is to not have anything specific to changesets, so I can remove and add only for my own peace of mind while testing